codec-http/src/test/java/io/netty/handler/codec/http/multipart/HttpPostMultiPartRequestDecoderTest.java

```diff
@@ -358,4 +358,121 @@ private void commonNotBadReleaseBuffersDuringDecoding(HttpDataFactory factory, b
         factory.cleanAllHttpData();
         decoder.destroy();
     }
+
+    // Issue #11668
+    private void commonTestFileDelimiterLFLastChunk(HttpDataFactory factory, boolean inMemory)
+            throws IOException {
+        int nbChunks = 2;
+        int bytesPerChunk = 100000;
+        int bytesLastChunk = 10000;
+        int fileSize = bytesPerChunk * nbChunks + bytesLastChunk; // set Xmx to a number lower than this and it crashes
+
+        String delimiter = "--861fbeab-cd20-470c-9609-d40a0f704466";
+        String prefix = delimiter + "\n" +
+                        "Content-Disposition: form-data; name=\"image\"; filename=\"guangzhou.jpeg\"\n" +
+                        "Content-Type: image/jpeg\n" +
+                        "Content-Length: " + fileSize + "\n" +
+                        "\n";
+
+        String suffix = "--861fbeab-cd20-470c-9609-d40a0f704466--";
+        byte[] bsuffix = suffix.getBytes(CharsetUtil.UTF_8);
+        byte[] bsuffixReal = new byte[bsuffix.length + 2];
+        for (int i = 0; i < bsuffix.length; i++) {
+            bsuffixReal[1 + i] = bsuffix[i];
+        }
+        bsuffixReal[0] = HttpConstants.LF;
+        bsuffixReal[bsuffixReal.length - 1] = HttpConstants.CR;
+        byte[] lastbody = {HttpConstants.LF};
+
+        HttpRequest request = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST, "/upload");
+        request.headers().set("content-type", "multipart/form-data; boundary=861fbeab-cd20-470c-9609-d40a0f704466");
+        // +4 => 2xCRLF (beginning, end)
+        request.headers().set("content-length", prefix.length() + fileSize + suffix.length() + 4);
+
+        HttpPostMultipartRequestDecoder decoder = new HttpPostMultipartRequestDecoder(factory, request);
+        ByteBuf buf = Unpooled.wrappedBuffer(prefix.getBytes(CharsetUtil.UTF_8));
+        decoder.offer(new DefaultHttpContent(buf));
+        assertNotNull(((HttpData) decoder.currentPartialHttpData()).content());
+        buf.release();
```

Comment:
I am a bit confused... Why we release here ? Shouldn't the decoder have ownership of the buffer now ?

Reply:
As it is out of any "Netty Handler" (as `SimpleChannelInboundHandler`), the test shall do as a normal handler acts, implying that buffer is released as the handler shall do.
The decoder takes ownership but on a copy, not the original one.
So, AFAIK, there should not be any confusion there.

Reply:
The previous change done on the decoder was about the "copied" buffer, not the original one. So it was a good catch since this buffer (copied from original) could be lost while not released.

Reply:
If you're OK, please resolve these points ;-)

Reply:
Perhaps it would be clearer if the test kept around variables with the `DefaultHttpContent` and released _that_, instead of the buffer. The `DefaultHttpContent` _is_ meant to own its buffer.

Reply:
I can try to change to this way (releasing `DefaultHttpContent`) to not introduce incomprehension (I shall then correct "all" tests, not only this one).

Reply:
Change done to `DefaultHttpContent.release()` instead to ensure comprehension.